src/macaron/repo_finder/repo_finder_deps_dev.py

```diff
@@ -177,61 +167,80 @@ def _create_urls(self, purl: PackageURL) -> tuple[list[str], RepoFinderInfo]:
+                return None, False
             purl = latest_purl
+            if not purl.version:
+                # Should be unreachable.
+                return None, False
 
-        return [f"{DepsDevRepoFinder.BASE_URL}{encode(str(purl), safe='')}"], outcome or RepoFinderInfo.FOUND
-
-    def _retrieve_json(self, url: str) -> str:
-        """
-        Attempt to retrieve the json file located at the passed URL.
-
-        Parameters
-        ----------
-        url : str
-            The URL for the GET request.
-
-        Returns
-        -------
-        str :
-            The retrieved file data or an empty string.
-        """
-        response = send_get_http_raw(url, {})
-
-        if not response:
-            return ""
-
-        return response.text
+        api_endpoint = DepsDevService.get_endpoint(
+            purl=False, path="/".join(["systems", purl.type, "packages", purl.name, "versions", purl.version])
+        )
+        target_url = urllib.parse.urlunsplit(api_endpoint)
+
+        result = send_get_http(target_url, headers={})
+        if not result:
+            return None, False
+
+        result_attestations = json_extract(result, ["attestations"], list)
+        if not result_attestations:
+            logger.debug("No attestations in result.")
+            return None, False
+        if len(result_attestations) > 1:
+            logger.debug("More than one attestation in result: %s", len(result_attestations))
+
+        attestation_url = json_extract(result_attestations, [0, "url"], str)
+        if not attestation_url:
+            logger.debug("No attestation reported for %s", purl)
+            return None, False
+
+        attestation_data = send_get_http(attestation_url, headers={})
+        if not attestation_data:
+            return None, False
+
+        bundle = json_extract(attestation_data, ["attestation_bundles"], list)
+        if not bundle:
+            logger.debug("No attestation bundle in response.")
+            return None, False
+        if len(bundle) > 1:
+            logger.debug("Bundle length greater than one: %s", len(bundle))
+
+        attestations = json_extract(bundle[0], ["attestations"], list)
+        if not attestations:
+            logger.debug("No attestations in response.")
+            return None, False
+        if len(attestations) > 1:
+            logger.debug("More than one attestation: %s", len(attestations))
+
+        if not isinstance(attestations[0], dict):
+            logger.debug("Attestation invalid.")
+            return None, False
+
+        return attestations[0], json_extract(result_attestations, [0, "verified"], bool) or False
 
-    def _read_json(self, json_data: str) -> tuple[list[str], RepoFinderInfo]:
+    @staticmethod
+    def extract_links(json_data: dict) -> tuple[list[str], RepoFinderInfo]:
         """
-        Parse the deps.dev json file and extract the repository links.
+        Extract the repository links from  the deps.dev json data.
```

Comment:
I have a naive question. Does the BASE endpoint and the PURL endpoint returns the JSON data with similar schema and content ?

Reply:
They return slightly different results in terms of the schema.
https://api.deps.dev/v3alpha/purl/pkg%3Anpm%2F%40colors%2Fcolors@1.4.0
https://api.deps.dev/v3alpha/systems/npm/packages/%40colors%2Fcolors/versions/1.4.0

Reply:
Ah, I see. Thanks